.github/workflows/gradle.yml

```diff
@@ -0,0 +1,74 @@
+name: continuous-integration
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-on-linux-jdk16:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 16
+        uses: actions/setup-java@v2
+        with:
+          java-version: '16 '
+          distribution: 'adopt'
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+      - name: Build with Gradle and generate the artifacts
+        run: ./gradlew assemble test
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Package
+          path: build/libs
```

Comment:
Do we need to remove these Gradle files from the cache as a final step in the build?
```
~/.gradle/caches/modules-2/modules-2.lock
~/.gradle/caches/modules-2/gc.properties
```

I noticed a [comment about it](https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle#caching-dependencies) in the GitHub documentation.

The first file was also removed by the previous Travis build.

Reply:
> Do we need to remove these Gradle files from the cache as a final step in the build?
> 
> ```
> ~/.gradle/caches/modules-2/modules-2.lock
> ~/.gradle/caches/modules-2/gc.properties
> ```
> 
> I noticed a [comment about it](https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle#caching-dependencies) in the GitHub documentation.
> 
> The first file was also removed by the previous Travis build.

hmm yeah, we would probably need to delete those. I'll add that step.